src/pages/Topic/Topic.tsx

```diff
@@ -86,17 +105,43 @@ export const Topic = ({ useTopic = _useTopic }: TopicProps): JSX.Element => {
     mapNodes,
     navigate,
     setInitialNodes,
-    setInitialEdges
+    setInitialEdges,
+    learningPathElementStatus
   ])
 
-  return initialNodes && initialEdges ? (
+  // On Close of IFrameModal, fetch new LearningPathElementStatus, update it in
+  // the persisted store and return it. Then close the IFrameModal and rerender page.
+  // Catch for getUser is handled in the useEffect
+  const handleOwnClose = () => {
+    getUser()
+    .then((user) => {
+      getLearningPathElementSpecificStatus(courseId, user.lms_user_id, lmsId)
+      .then((data) => {
+        setLearningPathElementSpecificStatus(courseId?.toString(), user.lms_user_id, data[0]).then((data) => {
+          setLearningPathElementStatus(data)
+          return handleClose()
+        })
+      })
+       .catch((error: string) => {
+        addSnackbar({
+          message: error,
+          severity: 'error',
+          autoHideDuration: 3000
+        })
+        return handleClose()
+      })
+    })
+  }
```

Comment:
Find den Namen etwas unpassend.
Wenn man das return an das Ende macht, spart man sich eins.

```suggestion
  const getHandleClose = () => {
    getUser()
    .then((user) => {
      getLearningPathElementSpecificStatus(courseId, user.lms_user_id, lmsId)
      .then((data) => {
        setLearningPathElementSpecificStatus(courseId?.toString(), user.lms_user_id, data[0]).then((data) => {
          setLearningPathElementStatus(data)
          return
        })
      })
       .catch((error: string) => {
        addSnackbar({
          message: error,
          severity: 'error',
          autoHideDuration: 3000
        })
        return
      })
    })
    return handleClose()
  }
```